### DISCUSSION TITLE
Removing points from a point cloud relative from another cloud.

### DISCUSSION THREAD

**Discussion by surmitbhui** [isl-org/Open3D#3976] [Q&A]
Hi, 
I am working with a point cloud data and I have access to two point clouds - one with all the ground plane points, and the other with all the elevated points. I am trying to find a way to filter the points in the elevated point cloud of less than 2m relative to the ground. I tried numpy filtering using `np.where` but it doesn't give the right result if I filter using a common z-axis threshold as the ground is not entirely a flat plane (max value in the z axis for the ground is 6m). Would be great if I could get some help. 

This is how I was trying to filter it currently. 

```python
elevation_extraction = non_ground_cloud.select_by_index(np.where(np.asarray(non_ground_cloud.points) < 2)[0])
```

**Comment by hduonggithub** [comment]
What you do is  wrong.

You did not select non-ground points that is < 2m from the ground points, you just selected non-ground points <2m. As a sequence, you will miss a point that has ground at, e.g. 3m and non-ground at 4m which is still less than 2m relative distance.
You have to subtract non-ground and ground if they are same no. of points ( I guess not)

You have to find corresponding points between ground and non-ground, by xy values only and compute distances between them. Just select points that has distance < threshold

When you have corresponding point set, substract them and select by numpy.where

**Comment by surmitbhui** [comment]
@hduonggithub thanks for your reply. You are right,  my approach was wrong, so I tried using the `pcd.compute_point_cloud_distance()` method from the Open3D tutorials. As far as I could tell from the documentation, it calculates the distance of the target point cloud using KDTree nearest neighbors approach. When you suggest to find corresponding points by xy value - do you mean to compute this distance and select the corresponding indices? I tried to revise my implementation through: 

```python
   dists = non_ground_cloud.compute_point_cloud_distance(ground_cloud)
   dists = np.asarray(dists)
   ind = np.where(dists < 2)[0]
   elev = non_ground_cloud.select_by_index(ind)
```
But I am not sure if this still is the right approach as when I implement this, even though the results are a lot better than the previous approach but still seems a bit off (There is a lot of regions with no points at all where there should theoretically be points) . Would be great if you could point me towards how to find corresponding points between the two clouds?

Sorry but I am a bit new to working with point cloud data and Open3d. Any help would be great!

**Reply by hduonggithub** [reply]
Go and check why some regions have no points to find out more

**Comment by germanros1987** [comment]
In order to provide some help, we would need to have more information about the point clouds (perhaps some examples?), and also access to the results.
